### PR TITLE
Guards in Stringutils

### PR DESCRIPTION
**Pull request by harshad1**
This PR addresses issues discussed here: https://github.com/gsantner/markor/pull/1002

Additional bounds checks have been added to StringUtils

Issues:
- [x] Reproduce the crash
- [x] Confirm that the crash is solved

### CODE REVIEW

**Comment by gsantner** [comment]
ready to merge?

**Comment by harshad1** [comment]
> thx , ready to merge?

I want to test for another ~1 hr. Will post here when I am done.

File: app/src/main/java/net/gsantner/markor/format/markdown/MarkdownAutoFormat.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public OrderedListLine getLevelStart() {
 
         public OrderedListLine getParent() {
             OrderedListLine line = null;
-            if (isEmpty || (!isTopLevel && lineStart > INDENT_DELTA)) {
+            if ((lineStart > 0) && (isEmpty || (!isTopLevel && lineStart > INDENT_DELTA))) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/format/markdown/MarkdownAutoFormat.java:152]: This fixes the original issue

**Comment by harshad1** [comment]
I think this is good to go

**Comment by gsantner** [comment]
k thx merging

**Comment by gsantner** [comment]
btw, overwritten the commit on master with squash of everything

**Pull request merged**